client/src/store/newsFeed/actions.ts

```diff
@@ -0,0 +1,70 @@
+import { AuthAPI } from '../api';
+import { showAlert } from '../alert/actions';
+import { Dispatch } from 'redux';
+import {
+    GET_NEWS_FEED_PENDING,
+    GET_NEWS_FEED_SUCCESS,
+    GET_MORE_NEWS_FEED_SUCCESS,
+    UPLOAD_NEXT_FEED_POSTS,
+    ALL_NEWS_FEED_LOADED,
+    CLEAR_NEWS_FEED_LOADED,
+} from './actionTypes';
+import { INewsFeed } from './reducers';
+
+export const getNewsFeedPending = (): { type: string } => ({
+    type: GET_NEWS_FEED_PENDING,
+})
+
+export const getNewsFeedSuccess = (userNews: INewsFeed): { type: string, payload: any } => ({
+    type: GET_NEWS_FEED_SUCCESS,
+    payload: userNews,
+})
+
+export const getMoreNewsFeedSuccess = (userNews: any): { type: string, payload: any } => ({
+    type: GET_MORE_NEWS_FEED_SUCCESS,
+    payload: userNews,
+})
+
+export const addNextFeedPosts = (page: number): {type: string, payload: number} => ({
+    type: UPLOAD_NEXT_FEED_POSTS,
+    payload: page,
+});
+
+export const clearNewsFeedLoaded = (): { type: string } => ({
+    type: CLEAR_NEWS_FEED_LOADED,
+})
+
+export const allNewsFeedLoaded = (): { type: string } => ({
+    type: ALL_NEWS_FEED_LOADED,
+})
+
+export const getNewsFeedAsync = (): (dispatch: Dispatch) => Promise<void> =>
+    async (dispatch: Dispatch): Promise<void> => {
+        try {
+            dispatch(getNewsFeedPending());
+            const res = await AuthAPI.get('/feed/1');
```

Comment:
why 1?

Reply:
Because its first page

Reply:
Where we can see 9 posts and after scroll we have lzy loading for next page